### PR TITLE
Automatically generated logo for the cheatsheets

### PR DESCRIPTION
**Pull request by jimustafa**
As has been discussed in #32, there is a need to keep the `matplotlib` logo in the cheatsheets up to date with the correct version in the tag line. This PR does a couple of things:

- upgrades and pins the `matplotlib` version, updates the dependencies, and simplifies the requirements specification
- adds a script for generating the `matplotlib` logo, originally written by @timhoffm
- modifies the script to include "Cheat sheets" and the `matplotlib` version in the tag line

There may be some small tweaks to make the logo more closely resemble the one by @rougier; well, maybe also a big tweak, as the font is currently Carlito, not Calibri.

### CODE REVIEW

**Comment by rougier** [comment]
Thanks. I don't think we need to tweak the logo. How different are the two logos (with Carlito and Calibri)?

**Comment by jimustafa** [comment]
Hey @rougier. Have you been able to inspect the build artifacts generated with the GitHub Actions workflow? The cheatsheets build for this PR incorporates the generated logo.

**Comment by rougier** [comment]
I've built locally the cheatsheets and handouts and they look ok to me. I did not test yet your PR.
Also, we need to modify the copyright notice (replace my name by Matplotlib)

**Comment by jimustafa** [comment]
OK, I added a few commits, to do the following:
- fix the requirements `Makefile`
- update the `matplotlib` version tag in the handouts
- configure [`bump2version`](https://github.com/c4urself/bump2version) to automatically update the `matplotib` version string in the necessary files when the `matplotlib` dependency is upgraded

Regarding the copyright notice... It seems that it should read `Copyright (c) 2021 Matplotlib Development Team` based on the license file in the `matplotib` repo. Is this, in fact, how it should be done?

Another question... There appears to be an inconsistency between the license file in this repo (BSD2) and the license statements in the cheatsheets and handouts (CC-BY); maybe that is fine. However, there are also different licenses for the cheatsheets and handouts, with some released under `CC-BY 4.0` and other under `CC-BY International 4.0`. Additionally, the cheatsheets do not acknowledge support from `NumFocus Grant #12345`. Can these be normalized?

**Comment by jimustafa** [comment]
Good day @rougier. I updated the copyright notices and normalized the licensing in a manner consistent with my previous comment. Please review and let me know what, if anything, needs to be changed. Thanks!

File: handout-beginner.tex [hunk 1]
```diff
@@ -296,9 +296,12 @@ \subsection*{\rmfamily Save \mdseries (bitmap or vector format)}
...
 \vfill
 %
-{\scriptsize Matplotlib 3.2 handout for beginners. Copyright (c)
-  2020 Nicolas P. Rougier. Released under a CC-BY International 4.0
-  License. Supported by NumFocus Grant \#12345.\par}
+{\scriptsize
+  Matplotlib 3.4.2 handout for beginners.
+  Copyright (c) 2021 Matplotlib Development Team.
+  Released under a CC-BY 4.0 International License.
+  Supported by NumFocus Grant \#12345.
```
Comment (rougier) [inline handout-beginner.tex:303]: Grant number was random actually. Maybe "Supported by NumFocus" or "with support by NumFocus" would do (or even nothing?)

File: handout-intermediate.tex [hunk 2]
```diff
@@ -196,9 +197,12 @@ \subsection*{\rmfamily Size \& DPI}
...
 \vfill
 %
-{\scriptsize Matplotlib 3.2 handout for intermediate users. Copyright
-  (c) 2020 Nicolas P. Rougier. Released under a CC-BY 4.0
-  License. Supported by NumFocus Grant \#12345.\par}
+{\scriptsize
+  Matplotlib 3.4.2 handout for intermediate users.
+  Copyright (c) 2021 Matplotlib Development Team.
+  Released under a CC-BY 4.0 International License.
+  Supported by NumFocus Grant \#12345.
```
Comment (rougier) [inline handout-intermediate.tex:204]: Same as previous comment, grant number should be removed.

File: handout-tips.tex [hunk 3]
```diff
@@ -242,10 +242,12 @@ \subsection*{\rmfamily Read the documentation}
...
 %
-{\scriptsize Matplotlib 3.2 handout for tips \& tricks.
-             Copyright (c) 2020 Nicolas P. Rougier.
-             Released under a CC-BY 4.0 License.
-             Supported by NumFocus Grant \#12345.\par}
+{\scriptsize
+  Matplotlib 3.4.2 handout for tips \& tricks.
+  Copyright (c) 2021 Matplotlib Development Team.
+  Released under a CC-BY 4.0 International License.
+  Supported by NumFocus Grant \#12345.
```
Comment (rougier) [inline handout-tips.tex:249]: Random gran number.

**Comment by rougier** [comment]
Looks good to me but the grant number (my fault). After that I think we can merge.

**Comment by jimustafa** [comment]
Thanks @rougier for the review. I removed the grant number placeholder, and corrected the capitalization of NumFOCUS.

**Comment by rougier** [comment]
Many, many thanks. I think we're good to go!

**Comment by timhoffm** [comment]
A major problem with  autogenerating the logo is that the font Calibri is no available. Thus the logo has the wrong font, which is a major branding issue.

I think the correct solution would be to take the logo as is and overlay the "Cheat Sheet" and version tests in LaTeX.

**Comment by story645** [comment]
The "grant #" was a Numfocus Small Development grant , if you want something to acknowledge.

**Pull request merged**